tests/unit/test_main.py

```diff
@@ -166,34 +167,173 @@ async def test_get_flows_to_be_installed(self, *args):
                         "instructions": [
-                            {"instruction_type": "goto_table", "table_id": 3}
+                            {"instruction_type": "goto_table", "table_id": 2}
                         ],
-                        "match": {"in_port": 1},
                     },
                 }
             ],
-            "id": "mocked_pipeline",
-            "status": "enabling",
+        }
+
+        # No flows installed
+        flows_by_switch = {}
+        self.napp.get_miss_flows = MagicMock()
+        self.napp.get_miss_flows.return_value = ({}, set())
+        self.napp.manage_miss_flows(pipeline, flows_by_switch)
+        assert mock_install.call_count == 0
+        assert mock_delete.call_count == 0
+        assert self.napp.get_miss_flows.call_count == 0
+
+        # No miss flows installed
+        flows_by_switch = {"00:00:00:00:00:00:00:01": [{"flow": {"owner": "of_core"}}]}
+        self.napp.get_miss_flows.return_value = ({}, set())
+        self.napp.manage_miss_flows(pipeline, flows_by_switch)
+        expected_arg = {
+            1: {
+                "priority": 0,
+                "instructions": [{"instruction_type": "goto_table", "table_id": 2}],
+            }
+        }
+        assert mock_install.call_count == 1
+        args = mock_install.call_args[0]
+        assert args[0] == expected_arg
+        assert args[1] == expected_arg.keys()
+        assert mock_delete.call_count == 0
+        assert self.napp.get_miss_flows.call_count == 1
+
+        # With miss flows installed
+        pipeline = {
+            "multi_table": [
+                {
+                    "table_id": 1,
+                    "table_miss_flow": {
+                        "priority": 0,
+                        "instructions": [
+                            {"instruction_type": "goto_table", "table_id": 2}
+                        ],
+                    },
+                },
+                {
+                    "table_id": 2,
+                    "table_miss_flow": {
+                        "priority": 0,
+                        "instructions": [
+                            {"instruction_type": "goto_table", "table_id": 5}
+                        ],
+                    },
+                },
+                {
+                    "table_id": 5,
+                    "table_miss_flow": {"priority": 99, "instructions": [{}]},
+                },
+            ]
+        }
+        dpid = "00:00:00:00:00:00:00:01"
+        flows_by_switch = {dpid: []}
+        for table_id in range(4):
+            flow = {
+                "flow": {
+                    "owner": "of_multi_table",
+                    "priority": 0,
+                    "instructions": [
+                        {"instruction_type": "goto_table", "table_id": table_id + 1}
+                    ],
+                    "table_id": table_id,
+                }
+            }
+            flows_by_switch[dpid].append(flow)
```

Comment:
Excellent that this cover both installs and deletions, could you also cover another case here when no changes (or additional calls to `mock_install` or `mock_delete`) are expected? I'm not sure if this has been indirectly covered though.

Reply:
It was not. I divided all tests for method into different methods since it was getting too large.